### PR TITLE
Fix any istype

### PR DESCRIPTION
**Pull request by galou**

### CODE REVIEW

**Comment by facontidavide** [comment]
Motivation?

**Comment by galou** [comment]
I need to convert blackboard entries to my Lua environment. I cannot use `tryCast` because a lot of things cast to a lot of things. `isType<bool>()` and similar returned `false` because of the type casting  inside `Any` (cf. unit-test). The second commit fixes this.

File: include/behaviortree_cpp/utils/safe_any.hpp [hunk 1]
```diff
@@ -269,15 +276,15 @@ inline Any &Any::operator =(const Any &other)
 
 inline bool Any::isNumber() const
 {
-  return _any.type() == typeid(int64_t) ||
-         _any.type() == typeid(uint64_t) ||
-         _any.type() == typeid(double);
+  return type() == typeid(int64_t) ||
+         type() == typeid(uint64_t) ||
+         type() == typeid(double);
```
Comment (facontidavide) [inline include/behaviortree_cpp/utils/safe_any.hpp:281]: Not sure about this. Shouldn't you include all the possible arithmetic types?

**Review by facontidavide — CHANGES_REQUESTED** [review]

File: include/behaviortree_cpp/utils/safe_any.hpp (hunk 1 above)
Comment (galou) [reply include/behaviortree_cpp/utils/safe_any.hpp:281]: Or use `_any.type()`?

Now the `isNumber` function is not called in behaviortree_cpp (`isCastedAsNumber` is used instead).

**Comment by facontidavide** [comment]
this is still public API, I want to be sure that the behavior of public method is not changed

**Comment by facontidavide** [comment]
I don't want to introduce the method isCastedType, therefore I cherry picked the other changes.

Thanks!

**Pull request closed without merging**